### PR TITLE
Requests for Sitemaps that don't exist return 404 error messages

### PR DESCRIPTION
**Pull request by mdbitz**
Resolution of https://github.com/Automattic/msm-sitemap/issues/79. Sitemap template is updated so that wp_die explicitly returns a 404 response code. In addition when building individual sitemap xml we explicitly return false when no data exists for the requested date. This ensures that 404 is returned instead of empty XML.

### CODE REVIEW

**Comment by mjangda** [comment]
This probably needs to be rebased against master as well.

File: msm-sitemap.php [hunk 1]
```diff
@@ -673,8 +673,7 @@ public static function build_individual_sitemap_xml( $year, $month, $day ) {
 			return $sitemap_content;
 		} else {
```
Comment (mjangda) [inline msm-sitemap.php]: We can kill the `else` clause here.

File: templates/full-sitemaps.php [hunk 2]
```diff
@@ -12,13 +12,17 @@
...
 $build_xml = Metro_Sitemap::build_xml( array( 'year' => $req_year, 'month' => $req_month, 'day' => $req_day ) );
 
 if ( $build_xml === false ) {
-	wp_die( __( 'Sorry, no sitemap available here.', 'msm-sitemap' ) );
+	wp_die( 
+		__( 'Sorry, no sitemap available here.', 'msm-sitemap' ),
+		__( 'Sitemap Not Available', 'msm-sitemap' ),
+		array ( 'response' => 404 )
+	);
 } else {
```
Comment (mjangda) [inline templates/full-sitemaps.php]: Let's kill the `else` clause here too.

Comment (mdbitz) [inline templates/full-sitemaps.php]: Thanks Mo. getting these updated to conform to standards.

**Comment by mdbitz** [comment]
Hi Mo: I've updated those else statements per your feedback and did a rebase. However after a rebase git has me re pull the branch from my fork, so I might be doing something wrong there.

**Pull request merged**